app/handler/books.go

```diff
@@ -112,3 +128,17 @@ func DeleteBook(w http.ResponseWriter, r *http.Request) {
 	books = newSetOfBooks
 	json.NewEncoder(w).Encode(books)
 }
+
+func findOrFail(id string) (book Book, err error) {
+	for _, b := range books {
+		if b.ID == id {
+			book = b
+		}
+	}
+
+	if book.ID == "" {
+		err = fmt.Errorf("Book Not Found")
+	}
+
+	return book, err
```

Comment:
since you did the returns of errors, no need to return an error here

Reply:
yeah makes sense